### PR TITLE
fix for windows part with unsuccessful reinstall (AO-17821)

### PR DESCRIPTION
**Pull request by melord**
https://swicloud.atlassian.net/browse/AO-17821

### CODE REVIEW

**Review by tomekkolo — APPROVED** [review]

File: molecule/complex_configuration/tests/test_complex_configuration.py [hunk 1]
```diff
@@ -10,10 +12,15 @@
...
+@pytest.fixture
+def version(host):
+    return host.package(PACKAGE_NAME).version
+
+
+def test_solarwinds_snap_agent_package(host, version):
+    swisnap = host.package(PACKAGE_NAME)
     assert swisnap.is_installed
-    assert swisnap.version.startswith("3.")
+    assert version.startswith("3.") or version.startswith("4.")
```
Comment (tomekkolo) [inline molecule/complex_configuration/tests/test_complex_configuration.py:23]: why not parse and have  "is greater"

File: molecule/complex_configuration/tests/test_complex_configuration.py [hunk 2]
```diff
@@ -24,11 +31,15 @@ def test_swinsapd_service(host):
...
+    # checking default collector plugins: aosystem, processes, logs
+    assert len(host.process.filter(user=SOLARWINDS, comm="snap-plugin-col")) == 3
+    # checking default publisher plugins: publisher-appoptics, publisher-processes
     assert len(host.process.filter(user=SOLARWINDS, comm="snap-plugin-pub")) == 2
 
 
-def test_sockets(host):
+def test_sockets(host, version):
+    if version.startswith("4."):
+        pytest.skip("Using unix socket from version 4.x.x")
```
Comment (tomekkolo) [inline molecule/complex_configuration/tests/test_complex_configuration.py:42]: the default is /opt/SolarWinds/Snap/run/swisnapd.sock folder on linux and similar in Program Data on windows. You can check if file exists

File: molecule/complex_configuration/tests/test_complex_configuration.py (hunk 1 above)
Comment (melord) [reply molecule/complex_configuration/tests/test_complex_configuration.py]: done

File: molecule/complex_configuration/tests/test_complex_configuration.py (hunk 2 above)
Comment (melord) [reply molecule/complex_configuration/tests/test_complex_configuration.py]: done, for version 4.x.x and greater `/opt/SolarWinds/Snap/run/swisnapd.sock` file is checked

**Review by lootek — DISMISSED** [review]
LGTM

**Review by tomekkolo — APPROVED** [review]

**Pull request merged**